core/src/main/java/org/mule/expression/DefaultExpressionManager.java

```diff
@@ -278,25 +158,20 @@ public Object match(String token)
         }, expression);
     }
 
-    public boolean isExpression(String string)
+    @Override
+    public TypedValue evaluateTyped(String expression, MuleEvent event)
     {
-        return (string.contains(DEFAULT_EXPRESSION_PREFIX));
+        expression = removeExpressionMarker(expression);
+        return expressionLanguage.evaluateTyped(expression, event);
     }
 
     @Override
-    public TypedValue evaluateTyped(String expression, MuleMessage message)
+    public boolean isExpression(String string)
     {
-        expression = removeExpressionMarker(expression);
-        return expressionLanguage.evaluateTyped(expression, message);
+        return (string.contains(DEFAULT_EXPRESSION_PREFIX));
```

Comment:
probably a bigger story, but might be good to have a better algorithm here. I think the TemplateParser can help with that